### PR TITLE
wifischedule: Add `ignore_stations` parameter.

### PR DESCRIPTION
**Pull request by bedaes**
Hi!

First thanks for this package it is really helpful.

I have added a simple way to ignore associated stations in the soft_disable_wifi function. I use this to turn the wifi off at night whether or not my Roomba is connected.

My goal is to get this change back into the openwrt/packages repo. Since you are the maintainer I thought it would best to get in touch with you first. :)

Also, I have no clou on how to build a package for OpenWRT though I'm pretty sure I didn't break anything, but maybe you can help me out with this?

I've also forked the openwrt/packages repo an added the commit there:
https://github.com/bedaes/packages/commit/8e451afe6840d6b805364a945860725f658c67e6

Do you have any thoughts on this?

Have a nice evening & best regards.

### CODE REVIEW

**Comment by newkit** [comment]
Hi, thanks for you interest in this package. As far as I understand, the feature you implemented will disconnect the WIFI, no matter if there are stations connected, correct? If so, this feature is already there, that's what the 'force' option is for. Does that work for you?

In regards to your questions on how to build I recommend you have a look here: [https://openwrt.org/docs/guide-developer/quickstart-build-images](https://openwrt.org/docs/guide-developer/quickstart-build-images)

In short terms - you need an appropriate linux system, the openWRT checkout, your changes and a bit of time ;-)

HTH,
Nils

**Comment by bedaes** [comment]
Hi, I think I should explain the feature based on a real world example. Let's say I have three devices A (Roomba), B (Smartphone) and C (Notebook). To keep this simple, the MACs of A, B and C are AA:..., BB:... and CC:.... The configuration looks like this:

```
config global
        option recheck_interval '10'
        option modules_retries '10'
        option enabled '1'
        option unload_modules '0'
        option ignore_stations 'AA:AA:AA:AA:AA:AA'
        option logging '1'

config entry 'Businesshours'
        option starttime '06:00'
        option forcewifidown '0'
        option enabled '1'
        option daysofweek 'Monday Tuesday Wednesday Thursday Friday Saturday Sun
        option stoptime '00:00'
```

Normally I would have either two options:

1. `option forcewifidown '0'`
keeps the wifi on until A, B and C are gone.

1. `option forcewifidown '1'`
force the wifi off no matter what.

But I'd like to have a third option:

3. `option forcewifidown '0'` + `option ignore_stations 'AA:AA:AA:AA:AA:AA'`
keep the wifi on till B and C are gone and completely ignore A.

Why do I want that? A is connected the whole time and therefore the wifi won't ever turn off. Also I do not want to forcefully kick my other devices B and C off the network at a fixed time.

Does this explanation make any sense to you?

@ how to build: I feared as much :)

**Comment by newkit** [comment]
Hehe, thanks for the explanation ;) makes perfectly sense to me. And thanks for contributing. Give me some time to check out and merge your changes, please. I'll let you know once it's upstream. :-D

**Comment by bedaes** [comment]
Hi there!

After nearly one year of use without a single issue I would say this change had enough real-world testing - maybe it's time to get this change merged? :)

**Comment by bedaes** [comment]
Any chance of getting the most recent version into https://github.com/openwrt/packages ?

**Comment by bedaes** [comment]
> Any chance of getting the most recent version into https://github.com/openwrt/packages ?

Maybe even before the MRs second birthday? :) @newkit
